RXPatchLib/RXPatcher.cs

```diff
@@ -17,13 +17,33 @@ public class RXPatcher
         const string DownloadSubPath = "download"; // Note that this directory will be automatically emptied after patching.
         const string TempSubPath = "apply"; // Note that this directory will be automatically emptied after patching.
 
-        public UpdateServerSelector Selector = null;
-        public string BaseURL = null;
+        //public UpdateServerSelector Selector = null;
+        public UpdateServerEntry BaseURL = null;
         public string WebPatchPath = null;
 
-        public async Task ApplyPatchFromWeb(string baseUrl, string targetPath, string applicationDirPath, IProgress<DirectoryPatcherProgressReport> progress, CancellationToken cancellationToken, string instructions_hash)
+        private static RXPatcher _instance;
+        public static RXPatcher Instance => _instance ?? (_instance = new RXPatcher());

```

Comment:
What's the rationale to changing this to a singleton? Have any exceptions/crashes been observed?

Otherwise this sounds fine, though I am curious if it might introduce any unintended race conditions.

Reply:
This was done really quite early on, I have yet to find any issues with making this instanced in this way, but it does make it easier to access the patcher when needed anywhere in code.